Api/REST.md

```diff
@@ -1,2 +1,218 @@
 ## REST
 
+We follow RESTFUL conventions in order to have a consistency between client and server communication. We are always using JSON as standard way of communicating for making the request and getting the response.
+
+Here are some guidelines for the best practices.
+
+
+
+### Routes
+
+Standard route format, for and API that returns costumers information, should be *`api/v1/customers`*.  We achieve this by using *`Route`* attribute and .NET routing keywords such as *`{{version:apiVersion}}`*, *`[controller]`* etc.
+
+Default route should be set ether on controller level or on base controller if controllers are inheriting it.
+
+
+
+```c#
+	[ApiController]
+    [Route("api/v{version:apiVersion}/[controller]")]
+    public class BaseApiController : ControllerBase
+    {
+        
+    }
+```
+
+
+
+### Versioning
+
+In order to provide long time support and migration, it is recommended to use versioning. For details of API versioning, please refer to [Versioning](Versioning.md) chapter.
+
+
+
+### Controllers
+
+API controllers should be used to separate actions by entity. Usually controllers are able to return array of entities, therefore they should be named in plural. Giving a meaningful and consistent names to controllers will result in cleaner code and cleaner routes. The name should be followed by suffix Controller as it is a keyword in .NET that is handled in routing.
+
+``` c#
+[ApiVersion("1.0")]
+public class CustomersController: BaseApiController
+{
+    
+}
+```
+
+In the given example, we are using *`BaseApiController`* inheritance to inherit the default route. Any routes added to controller or controller methods will be prefixed by route defined in *`BaseApiController`*. The route for this controller will be *`api/v1/customers`*.
+
+
+
+### Methods
+
+In .NET there are attributes for each RESTFUL method. Most common that are used are *`GET`*, *`POST`*, *`PUT`* and *`DELETE`*. Methods without a route will take a default route from the controller, but we can also add a route to the method it self.
+
+#### GET
+
+Get is a method reserved for getting information about a resource.
```

Comment:
Maybe worth mentioning that GET should not have any side effects (it can be called any number of times and data shouldn't change because of it).